SteamKit2/SteamKit2/Steam/Handlers/SteamMatchmaking/Lobby.cs

```diff
@@ -0,0 +1,383 @@
+            {
+                /// <summary>
+                /// String value to compare against.
+                /// </summary>
+                public string Value { get; }
+
+                /// <summary>
+                /// Initializes a new instance of the <see cref="StringFilter"/> class.
+                /// </summary>
+                /// <param name="key">The metadata key this filter pertains to.</param>
+                /// <param name="comparison">The comparison method used by this filter.</param>
+                /// <param name="value">String value to compare against.</param>
+                public StringFilter( string key, ELobbyFilterComparison comparison, string value ) : base( ELobbyFilterType.String, key, comparison )
+                {
+                    Value = value;
+                }
+
+                /// <summary>
+                /// Serializes the string filter into a representation used internally by SteamMatchmaking.
+                /// </summary>
+                /// <returns>A protobuf serializable representation of this filter.</returns>
+                public override CMsgClientMMSGetLobbyList.Filter Serialize()
+                {
+                    CMsgClientMMSGetLobbyList.Filter filter = base.Serialize();
+                    filter.value = Value;
+                    return filter;
+                }
+            }
+
+            /// <summary>
+            /// Can be used to filter lobbies by comparing a string against a value in each lobby's metadata.
+            /// </summary>
+            public sealed class Member
+            {
+                /// <summary>
+                /// SteamID of the lobby member.
+                /// </summary>
+                public SteamID SteamID { get; }
+
+                /// <summary>
+                /// Steam persona of the lobby member.
+                /// </summary>
+                public string PersonaName { get; }
+
+                /// <summary>
+                /// Metadata attached to the lobby member.
+                /// </summary>
+                public IReadOnlyDictionary<string, string> Metadata { get; }
+
+                internal Member( SteamID steamId, string personaName, IReadOnlyDictionary<string, string> metadata = null )
+                {
+                    SteamID = steamId;
+                    PersonaName = personaName;
+                    Metadata = metadata ?? new Dictionary<string, string>();
+                }
+
+                /// <summary>
+                /// Checks to see if this lobby member is equal to another. Only the SteamID of the lobby member is taken into account.
+                /// </summary>
+                /// <param name="obj"></param>
+                /// <returns>true, if obj is <see cref="Member"/> with a matching SteamID. Otherwise, false.</returns>
+                public override bool Equals( object obj )
+                {
+                    if ( obj is Member )
```

Comment:
Use `obj is Member member` to avoid a second type-check below (when casting).

Reply:
Ah yeah, sorry, this was running with my .NET 3.5 fork of SteamKit (https://github.com/Benjamin-Dobell/SteamKit/tree/net35) with an old Mono compiler not supporting the syntax. Will make the change 👍

Reply:
In theory you should be able to use this syntax and still target .NET 3.5... in practice setting up the SDKs like that can be a pain.

Reply:
Yeah, you're spot on.

I have managed to bump the language level a couple of times with my tool-chain. I can probably pull this off now and drastically clean up my (very hacky) backport.

My original issue was that I needed MDBs (Mono debug symbols), which requires using `xbuild` instead of `msbuild`. However, `xbuild` can't go above Language Level 6(?).

I ended up getting fed-up with the archaic tooling, swapped to `msbuild` and threw together a trivial Cecil-based CLI tool to convert the symbols from PDB to MDB (https://github.com/Benjamin-Dobell/pdbtomdb) - `pdb2mdb` doesn't handle the latest "portable" PDB file format. Ah, the joys of the ever changing .NET tooling ecosystem.

Reply:
`<DebugType>full</DebugType>` should get you older PDBs if you need.

Reply:
Ah, sorry, forgot to mention, I'm (generally) on macOS. Older PDBs are Windows only.